app/src/main/java/com/tyron/code/ui/iconmanager/IconManagerActivity.java

```diff
@@ -0,0 +1,432 @@
+package com.tyron.code.ui.iconmanager;
+
+import android.app.Activity;
+import android.content.ClipData;
+import android.content.ContentUris;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.Intent;
+import android.database.Cursor;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
+import android.graphics.drawable.GradientDrawable;
+import android.graphics.PorterDuff;
+import android.net.Uri;
+import android.text.TextUtils;
+import android.provider.DocumentsContract;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.ImageView;
+import android.widget.Toast;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+import android.os.Bundle;
+import androidx.appcompat.widget.Toolbar;
+import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+import com.tyron.code.ui.iconmanager.loader.LoaderDialog;
+import com.tyron.code.ui.iconmanager.util.FileUtil;
+import android.widget.Button;
+import androidx.recyclerview.widget.RecyclerView;
+import com.tyron.code.ui.iconmanager.adapter.IconAdapter;
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileOutputStream;
+import java.net.URLDecoder;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import com.tyron.builder.project.Project;
+import com.tyron.code.R;
+import com.tyron.code.ui.project.ProjectManager;
+import org.apache.commons.io.FileUtils;
+import com.tyron.common.util.AndroidUtilities;
+import com.google.android.material.dialog.MaterialAlertDialogBuilder;
+
+public class IconManagerActivity extends AppCompatActivity {
```

Comment:
Use fragments instead, CodeAssist follows the single activity architecture. (With one exception from the Settings activity)